vhost_user_fs/src/server.rs

```diff
@@ -63,7 +64,12 @@ impl<F: FileSystem + Sync> Server<F> {
     }
 
     #[allow(clippy::cognitive_complexity)]
-    pub fn handle_message(&self, mut r: Reader, w: Writer) -> Result<usize> {
+    pub fn handle_message(
+        &self,
+        mut r: Reader,
+        w: Writer,
+        vu_req: Option<&mut SlaveFsCacheReq>,
```

Comment:
As mentioned [here](https://github.com/cloud-hypervisor/cloud-hypervisor/issues/481#issuecomment-575226292), I think it'd be cleaner to define a trait to make things more generic. We don't really want the `server.rs` to know that it is dealing with `vhost-user` specifically.

Reply:
@sboeuf In order to avoid the cyclic dependency, I did the suggested trait inside vhost_rs crate instead of vhost_user_fs,  otherwise,

```
error: cyclic package dependency: package `vhost_rs v0.1.0 (/home/b.liu/1_source/3_alibaba/cloud-hypervisor/vhost_rs)` depends on itself. Cycle:
package `vhost_rs v0.1.0 (/home/b.liu/1_source/3_alibaba/cloud-hypervisor/vhost_rs)`
    ... which is depended on by `vm-virtio v0.1.0 (/home/b.liu/1_source/3_alibaba/cloud-hypervisor/vm-virtio)`
    ... which is depended on by `vhost_user_fs v0.1.0 (/home/b.liu/1_source/3_alibaba/cloud-hypervisor/vhost_user_fs)`
    ... which is depended on by `vhost_rs v0.1.0 (/home/b.liu/1_source/3_alibaba/cloud-hypervisor/vhost_rs)`
    ... which is depended on by `cloud-hypervisor v0.4.0 (/home/b.liu/1_source/3_alibaba/cloud-hypervisor)`
```